gdx/src/com/badlogic/gdx/utils/Base64Coder.java

```diff
@@ -23,73 +22,96 @@
  * Project home page: <a href="http://www.source-code.biz/base64coder/java/">www.source-code.biz/base64coder/java</a><br>
  * Author: Christian d'Heureuse, Inventec Informatik AG, Zurich, Switzerland<br>
  * Multi-licensed: EPL / LGPL / GPL / AL / BSD.
+ *
+ * @author Christian d'Heureuse
+ * @author vaxquis
  */
 
 package com.badlogic.gdx.utils;
 
 public class Base64Coder {
+	public static class CharMap {
+		protected final char[] encodingMap = new char[64];
+		protected final byte[] decodingMap = new byte[128];
+
+		public CharMap (char char63, char char64) {
+			int i = 0;
+			for (char c = 'A'; c <= 'Z'; c++) {
+				encodingMap[i++] = c;
+			}
+			for (char c = 'a'; c <= 'z'; c++) {
+				encodingMap[i++] = c;
+			}
+			for (char c = '0'; c <= '9'; c++) {
+				encodingMap[i++] = c;
+			}
+			encodingMap[i++] = char63;
+			encodingMap[i++] = char64;
+			for (i = 0; i < decodingMap.length; i++) {
+				decodingMap[i] = -1;
+			}
+			for (i = 0; i < 64; i++) {
+				decodingMap[encodingMap[i]] = (byte)i;
+			}
+		}
+
+		public byte[] getDecodingMap () {
+			return decodingMap;
+		}
+
+		public char[] getEncodingMap () {
+			return encodingMap;
+		}
+	}
 
 	// The line separator string of the operating system.
 	private static final String systemLineSeparator = "\n";
 
-	// Mapping table from 6-bit nibbles to Base64 characters.
-	private static char[] map1 = new char[64];
-	static {
-		int i = 0;
-		for (char c = 'A'; c <= 'Z'; c++)
-			map1[i++] = c;
-		for (char c = 'a'; c <= 'z'; c++)
-			map1[i++] = c;
-		for (char c = '0'; c <= '9'; c++)
-			map1[i++] = c;
-		map1[i++] = '+';
-		map1[i++] = '/';
-	}
-
-	// Mapping table from Base64 characters to 6-bit nibbles.
-	private static byte[] map2 = new byte[128];
-	static {
-		for (int i = 0; i < map2.length; i++)
-			map2[i] = -1;
-		for (int i = 0; i < 64; i++)
-			map2[map1[i]] = (byte)i;
-	}
+	public static final CharMap regularMap = new CharMap('+', '/'), urlsafeMap = new CharMap('-', '_');
 
 	/** Encodes a string into Base64 format. No blanks or line breaks are inserted.
-	 * 
 	 * @param s A String to be encoded.
 	 * @return A String containing the Base64 encoded data. */
 	public static String encodeString (String s) {
-		return new String(encode(s.getBytes()));
+		return encodeString(s, true);

```

Comment:
you're completely right; that defaults got there because I directly ported it from my personal code, and I was using url-safe as default in my lib. fixed.